Dockerfile

```diff
@@ -9,6 +9,9 @@ COPY requirements.txt /
 
 RUN pip3 install -r /requirements.txt
 
+# Get ST-Link drivers
+RUN apt-get install stlink-tools
```

Comment:
Why is this tool needed? I already added support for flashing and debugging the chip using openOCD. Lmk if you want me to show you how

Reply:
We were not able to flash the chip within the container using the St-link programmer. As a result, we add the drivers this way. Are you free today ? We could discuss it.

Reply:
this isn't necessary.
`scons --target=stw upload` to flash using openOCD
`scons --target=stw openocd-gdb` to debug using openOCD and GDB
The line you added to the docker-compose might still be necessary though. Can you test the above and let me know if that works? We will have to add openOCD to this repo at some point, like how we did for gcc

Reply:
We ran the command `scons--target=stw upload`  inside the container. It failed. To fix this issue, we added the stlink-tools to the container. Now the stw component can be successfully flashed to the device.

I suspect openocd-gdb will also cause issues since openocd is not installed in the container.

Reply:
Ahhhh shit yeah I just remembered that I _did_ use stlink tools to implement the upload command. Sorry my bad, I got confused. Yeah this looks fine, and yeah like you said we'll have to install openocd in the container as well. Fortunately that shouldn't be very difficult, can probably just `apt-get install openocd` in the dockerfile, but would be good to pin the versions of both the st-link tools as well as openocd. Can you guys work on that?

Reply:
Should we load openocd's stm32f103 config file in the script as default or should we let the user do that ?

Reply:
The tool I wrote (`site_scons/site_tools/openocd.py`) already allows the consumer to specify which config file to use. The `stw` SConscript file passes the correct config file to openocd